rudder-core/src/main/scala/com/normation/rudder/services/reports/ReportingServiceImpl.scala

```diff
@@ -167,29 +167,31 @@ trait CachedFindRuleNodeStatusReports extends ReportingService with CachedReposi
    * - initialized, else go find missing rule node status reports (one time for all)
    * - none reports is expired, else switch its status to "missing" for all components
    */
-  private[this] def checkAndUpdateCache(nodeIds: Set[NodeId]) : Box[Map[NodeId, Set[RuleNodeStatusReport]]] = this.synchronized {
-    if(nodeIds.isEmpty) {
+  private[this] def checkAndUpdateCache(nodeIdsToCheck: Set[NodeId]) : Box[Map[NodeId, Set[RuleNodeStatusReport]]] = this.synchronized {
+    if(nodeIdsToCheck.isEmpty) {
       Full(Map())
     } else {
       val now = DateTime.now
 
-      /*
-       * Three cases: 1/ cache does not exist, 2/ cache exists but expiration date expired, 3/ cache does note exists
-       * For simplicity (and keeping computation logic elsewhere than in a cache that already has its cache logic
-       * to manage), we will group 2 and 3, but it is well noted that it seems that a RuleNodeStatusReports that
-       * expired could be computed without any more logic than "every thing is missing".
-       *
-       */
-
-      val (expired, upToDate) = nodeIds.partition { id =>  //two group: expired id, and up to date info
-        cache.get(id) match {
-          case None      => true
-          case Some(set) => set.exists { _.expirationDate.isBefore(now) }
-        }
-      }
-
       for {
-        newStatus <- defaultFindRuleNodeStatusReports.findRuleNodeStatusReports(expired, Set())
+        allNodeIds        <- nodeInfoService.getAll.map( _.keySet)
+        //only try to update nodes that are accepted in Rudder
+        nodeIds            =  nodeIdsToCheck.intersect(allNodeIds)
+        /*
+         * Three cases: 1/ cache does not exist, 2/ cache exists but expiration date expired, 3/ cache does note exists
```

Comment:
Hmmm seems 1/ and 3/ are the same case (3 with typo ) one of them should be "cache does exists ? "